src/main/java/seedu/mealcompanion/command/recipe/RecipeDetailCommand.java

```diff
@@ -0,0 +1,54 @@
+package seedu.mealcompanion.command.recipe;
+
+import seedu.mealcompanion.MealCompanionException;
+import seedu.mealcompanion.MealCompanionSession;
+import seedu.mealcompanion.recipe.Recipe;
+
+//@@author ngyida
+/**
+ * Represents the "recipe detail" command.
+ */
+public class RecipeDetailCommand extends RecipeCommand {
+
+    String argument;
+    public RecipeDetailCommand(String argument) {
+        this.argument = argument;
+    }
+
+    /**
+     * Check if a string is a number
+     * @param argument the string to check for
+     * @return true if string is number, false otherwise
+     */
+    private boolean isNum(String argument) {
+        return argument.matches("[0-9]+");  //match a number with optional '-' and decimal.
+    }
+
+    @Override
+    public void execute(MealCompanionSession mealCompanionSession) {
+        try {
+            if (this.argument == null) {
+                throw new MealCompanionException("Missing argument!");
+            }
+
+            int index;
+
+            if (!isNum(this.argument)) {
+                index = mealCompanionSession.getRecipes().findIndex(this.argument);
```

Comment:
Should we allow users to use the command by index only? Or recipe name also can?